server/app.py

```diff
@@ -0,0 +1,551 @@
+        else:
+            print "ERROR!: Someone is trying to log in, but cxDocs wasn't started with login enabled."
+            print "If you'd like to enable login, you'll need to set up your postgres user database then run:"
+            print "$python start.py -u databaseUsername -p databasePassword"
+            print "Use $python start.py -h for more help. And check cxDocs.log for more helpful error messages."
+            return None
+
+    """ reads the config document to build a list of names of rules documents and their filepath.
+
+    returns a list of tuples, where the first field is the name of the document and the second is the
+        filepath to the document. Assumed to be the rules for the game Compound X from github.bleehu/Compound_X"""
+    def get_rules_docs():
+            rulesDocs = []
+            if config.has_option('Parser', 'basic_rules_filepath'):
+                rulesDocs.append(('Basic Rules','/docs/basic'))
+
+            if config.has_option('Parser', 'combat_rules_filepath'):
+                rulesDocs.append(('Combat Rules','/docs/combat'))
+
+            if config.has_option('Parser', 'damage_types_filepath'):
+                rulesDocs.append(('Damage Types and Armors','/docs/damagetypes'))
+
+            if config.has_option('Parser', 'conditions_filepath'):
+                rulesDocs.append(('Conditions','/docs/conditions'))
+
+            if config.has_option('Parser', 'level_up_filepath'):
+                rulesDocs.append(('Level Up Rules', '/docs/levelup'))
+
+            if config.has_option('Parser', 'cloaking_rules_filepath'):
+                rulesDocs.append(('Cloaking Rules','/docs/cloaking'))
+
+            if config.has_option('Parser', 'glossary_filepath'):
+                rulesDocs.append(('Glossary of Terms','/docs/glossary'))
+
+            if len(rulesDocs) < 1:
+                log.warn("Looked for Rules documents, but didn't find any. See config/README.md to configure rules docs.")
+                return None
+            return rulesDocs
+
+    def get_items_docs():
+        itemsDocs = []
+        if config.has_option('Parser', 'melee_weapons_filepath'):
+            itemsDocs.append(('Melee Weapons','/docs/meleeWeapons'))
+
+        if config.has_option('Parser', 'pistols_filepath'):
+            itemsDocs.append(('Pistols','/docs/pistols'))
+
+        if config.has_option('Parser', 'smgs_filepath'):
+            itemsDocs.append(('Submachine Guns','/docs/smgs'))
+
+        if config.has_option('Parser', 'carbines_filepath'):
+            itemsDocs.append(('Carbines and Assault Rifles','/docs/carbines'))
+
+        if config.has_option('Parser', 'long_rifles_filepath'):
+            itemsDocs.append(('Long Rifles and DMRs','/docs/longRifles'))
+
+        if config.has_option('Parser', 'machineguns_filepath'):
+            itemsDocs.append(('Machine Guns and Rocket Launchers','/docs/machineguns'))
+
+        if config.has_option('Parser', 'explosives_filepath'):
```

Comment:
Missing some weapon types?

Reply:
Several of the documents have two types of weapons in them each; such as SMGs and Shotguns.